features/telemetry/distributed/common/src/main/java/org/opennms/netmgt/telemetry/distributed/common/PropertyTree.java

```diff
@@ -159,6 +160,39 @@ public String getRequiredString(final String... path) {
                 .orElseGet(Collections::emptyMap);
     }
 
+    /**
+     * Get the values of all child nodes at the given path, enumerating any subtrees and converting their
+     * paths into dot-separated property names. This is intended to handle parameter keys containing dots -
+     * see NMS-12738.
+     *
+     * @param path the path of the node to return split into its elements
+     * @return the {@link Map} from the children node names to its values. If one of the nodes in the path does not exist this will return an empty {@link Map}
+     */
+
+    public Map<String, String> getMapR(final String... path) {
+        Map<String,String> outmap = new HashMap<String,String>();
+        Optional<Node> node = this.find(path);
+        if(node.isPresent()) {
```

Comment:
`node.ifPresent(...)` can be used to avoid the `node.get()` calls below and makes this more functional.

Reply:
Thanks! That's much more elegant.